File: tests/rosettacode/binsearch.go
package main

import "fmt"

func binarySearch(arr []int, target int) int {
	lo, hi := 0, len(arr)-1
	for lo <= hi {
		mid := (lo + hi) / 2
		if arr[mid] == target {
			return mid
		} else if arr[mid] < target {
			lo = mid + 1
		} else {
			hi = mid - 1
		}
	}
	return -1
}

func main() {
	arr := []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
	target := 7
	idx := binarySearch(arr, target)
	if idx >= 0 {
		fmt.Printf("Found %d at index %d\n", target, idx)
	} else {
		fmt.Printf("%d not found\n", target)
	}
}

File: tests/java-rosettacode/Binsearch.java
public class Binsearch {
    public static int binarySearch(int[] arr, int target) {
        int lo = 0, hi = arr.length - 1;
        while (lo <= hi) {
            int mid = lo + (hi - lo) / 2;
            if (arr[mid] == target) return mid;
            else if (arr[mid] < target) lo = mid + 1;
            else hi = mid - 1;
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] sorted = {1, 3, 5, 7, 9, 11, 13};
        int[] targets = {7, 8};
        for (int t : targets) {
            int idx = binarySearch(sorted, t);
            if (idx >= 0) {
                System.out.printf("%d found at index %d%n", t, idx);
            } else {
                System.out.printf("%d not found%n", t);
            }
        }
    }
}
